pyccel/codegen/printing/ccode.py

```diff
@@ -222,10 +234,14 @@ def function_signature(self, expr):
         if not expr.arguments:
             arg_code = 'void'
         else:
-            arg_dtypes = [self._print(i.dtype) for i in expr.arguments]
-            arg_dtypes = [dtype_registry[(dtype, arg.precision)] for dtype,arg in zip(arg_dtypes, expr.arguments)]
-            arguments  = [self._print(i) for i in expr.arguments]
-            arg_code   = ', '.join(dtype + ' ' + arg for dtype,arg in zip(arg_dtypes,arguments))
+            #to change for arguments with ranks * ** etc...
+            #arg_dtypes = [self._print(i.dtype) for i in expr.arguments]
+            #arg_dtypes = [dtype_registry[(dtype, arg.precision)] for dtype,arg in zip(arg_dtypes, expr.arguments)]
+            #arguments  = [self._print(i) for i in expr.arguments]
+            #arg_code   = ', '.join(dtype + ' ' + arg for dtype,arg in zip(arg_dtypes,arguments))
+            argument_declare = [Declare(i.dtype, i) for i in expr.arguments]
```

Comment:
Why are there declarations in the function signature? Does this not lead to semi colons in the wrong place?

Reply:
The line below removes the semi-colon and the new method avoids repetition

Reply:
I used declare in  function signature for args with rank greater than 1 exemple 
```
void *somefunc(int * x,   char * str)
```